### ISSUE TITLE
Dense layers after GAP

### ISSUE THREAD

**Issue by CalinTimbusLucian** [jacobgil/keras-cam#12]
Rather a question than an issue, but I was wondering whether a Dense layer in between the GAP and the final softmax activation layer could be added.
Thank you.

**Comment by CalinTimbusLucian** [comment]
Figured it out, thanks :)

**Comment by mammadjv** [comment]
I had a problem with the approach which is clear in this line:
https://github.com/jacobgil/keras-cam/blob/2b7ada2c5c819808a174444e82622315a07fa11e/model.py#L59
As it is mentioned in the paper, trained weights in this layer is used for weighted sum over the last produced activation maps.
As I know, for predicting a non-linear function and class score in MLP, there should be at-least two layers (one hidden layer and an output layer like Softmax).
But here, right after the GAP layer, only one FC layer with two units is added for classification.
Can anyone explain the reason?
